pipeline/recorder/src/test/scala/uk/ac/wellcome/platform/recorder/services/RecorderWorkerServiceTest.scala

```diff
@@ -74,25 +55,19 @@ class RecorderWorkerServiceTest
   }
 
   it("doesn't overwrite a newer work with an older work") {
-    val olderWork = createUnidentifiedWork
-    val newerWork = olderWork.copy(version = 10, title = "A nice new thing")
-
-    withLocalDynamoDbTable { table =>
-      withLocalSnsTopic { topic =>
-        withLocalS3Bucket { storageBucket =>
-          withLocalSqsQueue { queue =>
-            withLocalSnsTopic { topic =>
-              withWorkerService(table, storageBucket, topic, queue) { _ =>
-                sendMessage[TransformedBaseWork](queue = queue, newerWork)
-                eventually {
-                  assertStoredSingleWork(topic, table, newerWork)
-                  sendMessage[TransformedBaseWork](
-                    queue = queue,
-                    obj = olderWork)
-                  eventually {
-                    assertStoredSingleWork(topic, table, newerWork)
-                  }
-                }
+    withLocalSqsQueue { queue =>
+      withMemoryMessageSender { msgSender =>
+        withRecorderVhs { vhs =>
+          withWorkerService(queue, vhs, msgSender) { service =>
+            val olderWork = createUnidentifiedWork
+            val newerWork =
+              olderWork.copy(version = 10, title = "A nice new thing")
+            sendMessage[TransformedBaseWork](queue = queue, newerWork)
+            eventually {
+              assertWorkStored(vhs, newerWork)
+              sendMessage[TransformedBaseWork](queue = queue, obj = olderWork)
```

Comment:
Reading this test for the first time in a while, I wonder if we need a `Thread.sleep()` in here?

This test would pass immediately even if we do eventually overwrite the newer work – because it's the first thing we wrote, it's stored when we do the initial check, even if the worker is still processing the SQS message.

Reply:
V good catch! This actually high lighted a bug in the test:

```scala
assertWorkStored(vhs, newerWork)
```
Using the default `expectedVhsVersion = 0` arg this was testing that the older `newerWork` was stored.

Instead of adding a `sleep` I have done this instead:

```scala
eventually { assertQueueEmpty(queue) }
assertWorkStored(vhs, newerWork, 1)
```